common/etc/nginx/include/s3gateway.js

```diff
@@ -1041,10 +1041,25 @@ async function _fetchEC2RoleCredentials() {
 async function _fetchWebIdentityCredentials(r) {
     var arn   = process.env['AWS_ROLE_ARN'];
     var name  = process.env['HOSTNAME'] || 'nginx-s3-gateway';
-    var sts_endpoint = process.env['STS_ENDPOINT'] || 'https://sts.amazonaws.com';
+
+    var sts_endpoint = process.env['STS_ENDPOINT'];
+    if (!sts_endpoint) {
+        var sts_regional = process.env['AWS_STS_REGIONAL_ENDPOINTS'] || 'global';
```

Comment:
Could you please add docs for the `AWS_STS_REGIONAL_ENDPOINTS` environment variable as well?

Reply:
On second thought, there is no need to delaying merging on account of docs.